File: coding/15th May 2025/Kth Distinct String in an Array.java
//Kth Distinct String in an Array
class Solution 
{
    public String kthDistinct(String[] arr, int k) 
    {
        HashMap<String, Integer> map = new HashMap<>();
        for(String s : arr)
        {
            if(map.containsKey(s))
                map.put(s, map.get(s) + 1);
            else
                map.put(s, 1);
        }
        for(String s : arr)
        {
            if(map.get(s) == 1)
                k--;
            if(k == 0)
                return s;
        }
        return "";
    }
}

File: Coding in Python/27th January 2026/Kth Distinct String in an Array.py
#Kth Distinct String in an Array
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        map1 = {}
        for ele in arAr:
            if ele in map1:
                map1[ele] += 1
            else:
                map1[ele] = 1
        c = 0
        for ele in arr:
            if map1[ele] == 1:
                c += 1
            if c == k:
                return ele
        return ""
